### DISCUSSION TITLE
Old links do not show visit stats after upgrade 3.5 to 4.2 Error -> No URL found with short code "{insert code}"

### DISCUSSION THREAD

**Discussion by russkiimike** [shlinkio/shlink#2305] [Help wanted]
### Shlink version

Server: v4.2.3 - Client: v4.3.0

### PHP version

8.2.4

### How do you serve Shlink

Self-hosted RoadRunner

### Database engine

MySQL

### Database version

8.0

### Current behavior

Hey, I  upgraded my instance from 3.5 ish to the latest version and now my old shortlinks visit stats can not be viewed individually. I am getting the `No URL found with short code "{insert code}"`. You can still see the clicks are registered on the global visit stats (main page). Newly created links visti stats work fine. All the redirects also work fine. Can't access old links via CLI either

### Expected behavior

Expect to be able to view individual visit stats

### Minimum steps to reproduce

Update from v3.5 to v4.2
update php, mysql and start serving with roadrunner
change redirect domains

**Comment by acelaya** [comment]
A couple questions:

1. Are you using multiple domains?
2. Can you check if there's any entry in the `domains` table of your database?
3. Do your short URLs have a value in the `domain_id` column of the `short_urls` table?

**Reply by russkiimike** [reply]
Hey, thanks for the quick reply. I am using only one domain atm but I got two entries for `domains` also short URL's are NULL value for `domain_id`

**Reply by acelaya** [reply]
What value did you set as DEFAULT_DOMAIN?

**Reply by russkiimike** [reply]
I only have one domain that I use, other is dormant, I set it as default on the web app at app.shlink.io

**Reply by acelaya** [reply]
I'm not sure what you mean by that, but in any case, this seems like a configuration issue related with domains. 

In order to continue helping you I'm going to need you to put together a minimum reproducible environment via docker-compose, where you pass configuration as close to reality as possible.

**Reply by russkiimike** [reply]
I just don't understand the DEFAULT_DOMAIN , do you mean what value did I set it on the cli ? Overall ? Like changed the default domain on the web app when my previous got expired. It's set as 
![image](https://github.com/user-attachments/assets/bc5c1dfc-8654-4854-b776-b154e85f1f7e)

I assume it's because of missing domain_id fields and what not ?
![image](https://github.com/user-attachments/assets/7c3dc61d-8c0f-4559-b71e-a907b8177ad8)

**Reply by russkiimike** [reply]
Hello, any updates, I don't really know how to do docker compose environment ?

**Reply by russkiimike** [reply]
hi @acelaya can I get any help with this please ?

**Reply by russkiimike** [reply]
As per reproduction steps like I said , I performed the upgraded steps from the guide and then I changed to a new domain twice via the web app, that's all I did @acelaya
